### PR TITLE
Update Strapi to v5

### LINKED ISSUE #30: chore: update Strapi to v5
## Description

https://docs.strapi.io/dev-docs/upgrade-tool
https://docs.strapi.io/dev-docs/migration/v4-to-v5/step-by-step

### Strapi

- update to v5 version
- update plugins to work with v5 
- migrate code to use Documents Service API
- migrate lifecycles to v5
- verify the need for `patch-package` (because of nested components validation)
- update README

### UI

- update related API code - API fetch functions, `attributes` wrapper, `documentId` instead of `id` ...
- update API responses typing mechanism

### PR DESCRIPTION
**Pull request by tocosastalo**
## Description

- update Strapi v4 to Strapi v5 and migrate code
- update API related FE code
- update most of BE and FE dependecies
- remove Stripe plugin from repository. Stripe is no longer supported by default in this starter. We plan to update it,
move it to a public repository, and offer it as a plugin for the community
- update READMEs

closes #30

### CODE REVIEW

File: apps/ui/src/components/page-builder/components/BasicImage.tsx [hunk 1]
```diff
@@ -31,9 +31,11 @@ export function BasicImage({
 }: BasicImageProps) {
   removeThisWhenYouNeedMe("BasicImage")
 
-  const media: StrapiDataWrapper<StrapiImageMedia> = component?.media
+  component?.media
```
Comment (dominik-juriga) [inline apps/ui/src/components/page-builder/components/BasicImage.tsx]: Don't think this is useful

File: apps/ui/src/components/page-builder/components/BasicImage.tsx [hunk 2]
```diff
@@ -43,25 +45,27 @@ export function BasicImage({
...
+  const attributes = media
 
   return (
     <ImageComp
       style={{ ...sizes, ...imgProps.style }}
       className={className}
       // @ts-expect-error - src needs to be typed in better way
       src={attributes?.url}
-      fallbackSrc={component?.fallbackSrc}
+      fallbackSrc={component?.fallbackSrc ?? undefined}
```
Comment (dominik-juriga) [inline apps/ui/src/components/page-builder/components/BasicImage.tsx:64]: Is it necessary to be explicit here? The optional chaining should return undefined implicitly, even if `component` is null.

File: apps/ui/src/lib/next-helpers.ts [hunk 3]
```diff
@@ -91,7 +92,7 @@ export async function getMetadataFromStrapi({
       { omitAuthorization: true }
     )
 
-    const seo = res.data?.attributes.seo
+    const seo = (res.data as any)?.seo
```
Comment (dominik-juriga) [inline apps/ui/src/lib/next-helpers.ts]: Not entirely sure this is good, as we'll lose the typing for the entire SEO object. Was it necessary to remove the Extract type?

**Review by dominik-juriga — APPROVED** [review]
I've got just a couple of remarkd, but otherwise looks good to me!

File: apps/ui/src/components/page-builder/components/BasicImage.tsx (hunk 2 above)
Comment (tocosastalo) [reply apps/ui/src/components/page-builder/components/BasicImage.tsx:64]: The main reason is that in v5 the inferred types of frontend attributes have `... | undefined | null` type always. In v4 it was `... | undefined`. So it breaks things in many places where we expected just `undefined`

**Pull request merged**